Modules/posixmodule.c

```diff
@@ -321,29 +323,29 @@ corresponding Unix manual entries for more information on calls.");
 #  if defined(TIOCGWINSZ)
 #    define TERMSIZE_USE_IOCTL
 #  endif
-#endif /* MS_WINDOWS */
+#endif /* HAVE_WINDOWS_CONSOLE_IO */
 
 /* Various compilers have only certain posix functions */
 /* XXX Gosh I wish these were all moved into pyconfig.h */
 #if defined(__WATCOMC__) && !defined(__QNX__)           /* Watcom compiler */
 #  define HAVE_OPENDIR    1
 #  define HAVE_SYSTEM     1
 #  include <process.h>
-#else
-#  ifdef _MSC_VER
-     /* Microsoft compiler */
+#elif defined( _MSC_VER)
+  /* Microsoft compiler */
+#  ifdef MS_WINDOWS_DESKTOP
```

Comment:
For the CRT spawn/exec/system functions:

```suggestion
#  ifdef MS_WINDOWS_DESKTOP | MS_WINDOWS_SYSTEM
```

For `HAVE_GETPPID`, in the 10.0.19041 SDK, the process snapshot API is available for `APP` and `SYSTEM`. I suppose we're also to include `DESKTOP` explicitly for devs who don't know that `APP` is defined in that case.

For `HAVE_GETLOGIN`, `GetUserNameW()` is limited to `DESKTOP`.

Personal comment:

Neither `getppid()` nor `getlogin()` is actually called on Windows. They're unrelated to this section, i.e. "Various compilers have only certain posix functions". The same applies to `HAVE_PIPE`, since we implement `os.pipe()` directly via `CreatePipe()`.